src/socket.cr

```diff
@@ -391,7 +391,14 @@ class Socket < IO
   end
 
   def reuse_port?
-    getsockopt_bool LibC::SO_REUSEPORT
+    0 != getsockopt(LibC::SO_REUSEPORT, 0) do |errno|
```

Comment:
Ouch, that looks sooo weird... Please, assign the return value to a variable and compare with that:

```cr
ret = getsockopt(LibC::SO_REUSEPORT, 0) do |errno|
  # ...
end
ret != 0
```